graylog2-server/src/main/java/org/graylog2/shared/journal/LocalKafkaJournal.java

```diff
@@ -699,12 +711,15 @@ public List<JournalReadEntry> read(long readOffset, long requestedMaximumCount)
                 lastOffset = messageAndOffset.offset();
 
                 final byte[] payloadBytes = ByteBufferUtils.readBytes(messageAndOffset.message().payload());
-                if (LOG.isTraceEnabled()) {
+                if (LOG.isTraceEnabled() || includeMessageId) {
                     final byte[] keyBytes = ByteBufferUtils.readBytes(messageAndOffset.message().key());
+                    messages.add(new JournalReadEntry(keyBytes, payloadBytes, messageAndOffset.offset()));
                     LOG.trace("Read message {} contains {}", bytesToHex(keyBytes), bytesToHex(payloadBytes));
```

Comment:
We are now calling this also in cases where tracing is not enabled (but `includeMessageId` is `true`). Not a big problem, but I think also for readability it would be helpful to untangle the conditions a bit.